[quant][graphmode][fx] Guard the supported quantization type for add/mul

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52414 [qunat][graphmode][fx] Enable test for non quantized input for cat
* **#52413 [quant][graphmode][fx] Guard the supported quantization type for add/mul**
* #52412 [quant][graphmode][fx] Enable test for non quantized input for add/mul
* #52369 [quant][graphmode][fx] Fix fp16 dynamic quant for functional linear

Summary:
TODO: We'll need to add this guard for other ops as well

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_mul_add_fp16_config

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26503348](https://our.internmc.facebook.com/intern/diff/D26503348)

jerryzh168 committed:
[quant][graphmode][fx] Guard the supported quantization type for add/mul

Summary:
TODO: We'll need to add this guard for other ops as well

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_mul_add_fp16_config

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Guard the supported quantization type for add/mul"


Summary:
TODO: We'll need to add this guard for other ops as well

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_mul_add_fp16_config

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26503348](https://our.internmc.facebook.com/intern/diff/D26503348)

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantize_fx.py:
```diff
@@ -1592,6 +1592,38 @@ def forward(self, x):
+        m = prepare_fx(model, qconfig_dict)
+        m = convert_fx(m)
+        # make sure it runs
+        m(torch.randn(5, 5))
```
Please also add a check here to make sure the graph has the expected ops.

supriyar reviewed (approved):
looks good, left a comment about testing,.

jerryzh168 committed:
Update on "[quant][graphmode][fx] Guard the supported quantization type for add/mul"


Summary:
TODO: We'll need to add this guard for other ops as well

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_mul_add_fp16_config

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26503348](https://our.internmc.facebook.com/intern/diff/D26503348)

[ghstack-poisoned]
